storage_controller/src/background_node_operations.rs

```diff
@@ -0,0 +1,233 @@
+use std::{borrow::Cow, collections::HashMap, fmt::Debug, fmt::Display, sync::Arc};
+
+use tokio::{sync::mpsc::error::TrySendError, task::JoinHandle};
+use tokio_util::sync::CancellationToken;
+use utils::id::NodeId;
+
+use crate::service::Service;
+
+pub(crate) const MAX_RECONCILES_PER_OPERATION: usize = 10;
+
+#[derive(Copy, Clone)]
+pub(crate) struct Drain {
+    node_id: NodeId,
+}
+
+#[derive(Copy, Clone)]
+pub(crate) struct Fill {
+    node_id: NodeId,
+}
+
+pub(crate) enum Operation {
```

Comment:
Haven't we avoided the concept of `Operation` during the design of the storage controller, building it around the concept of reconciliation instead? Should we mix the two concepts in the storage controller?

Reply:
I pondered this in the Alternatives section of the [RFC](https://github.com/neondatabase/neon/pull/7704/files#diff-fce164fca2f54d865355baf59fdc27c0a3d1f417952913e282b5b013e5af81f2R321-R337).

Reply:
It's worth noting that these "operations" aren't entirely divorced from the idea of reconciliation. The underlying mechanism for applying changes is still reconciliation. Think of it like an explicitly controlled set of reconciliations.